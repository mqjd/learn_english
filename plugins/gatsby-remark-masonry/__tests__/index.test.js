import remarkParse from 'remark-parse';
import remarkStringify from 'remark-stringify';
import { unified } from 'unified';
import remarkGfm from 'remark-gfm';
import { createProcessor } from '@mdx-js/mdx';
const remarkMdx = require('remark-mdx');
import rehypeMetaAsAttributes from '@lekoarts/rehype-meta-as-attributes';

import plugin from '../';

const toHAST = require(`mdast-util-to-hast`);
const hastToHTML = require(`hast-util-to-html`);

const Remark = unified()
  .use(remarkParse)
  .use(remarkMdx)
  .use(remarkGfm)
  .use(remarkStringify)
  .freeze();

const remark = new Remark().data(`settings`, {
  commonmark: true,
  footnotes: true,
  pedantic: true,
});

const createNode = (content) => {
  const node = {
    id: 1234,
  };

  const markdownNode = {
    id: `${node.id} >>> MarkdownRemark`,
    children: [],
    parent: node.id,
    internal: {
      content,
      contentDigest: `some-hash`,
      type: `MarkdownRemark`,
    },
  };

  markdownNode.frontmatter = {
    title: ``, // always include a title
    parent: node.id,
  };

  return markdownNode;
};

const createPluginOptions = (content, getRemarkFileDependency = false) => {
  const ast = remark.parse(content);
  const dirName = 'not-real';
  return {
    files: [],
    markdownNode: ast,
    markdownAST: remark.parse(content),
    getNode: () => {
      return {
        dir: dirName,
      };
    },
    compiler: {
      parseString: remark.parse.bind(remark),
      generateHTML: (node) => hastToHTML(toHAST(node)),
    },
    getRemarkFileDependency,
  };
};

const md = `


{ /* table-style: masonry;theme:dark */ }

| Word          | Explain                                                                                                               |
| ------------- | --------------------------------------------------------------------------------------------------------------------- |
| taco            | A taco is a crispy Mexican pancake made from corn and eggs, which is folded and filled with meat, vegetables, and a spicy sauce. <br/>![taco](https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcTIi66HnFYBiXcxxJORXn089VWo9DfKxoVuYDNNcR6pE1GNPXVfCMUVc3RlqfO-ogztO1Y&usqp=CAU)                                                                                                                      |


`;

test(`gatsby-remark-masonry test`, async () => {
  const markdownAST = createProcessor({
    remarkPlugins: [remarkGfm],
    rehypePlugins: [rehypeMetaAsAttributes],
  }).parse(md);

  plugin({ markdownAST });

  console.log(markdownAST);
});
