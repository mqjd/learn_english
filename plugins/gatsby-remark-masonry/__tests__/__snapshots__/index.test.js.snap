// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`it transforms images in markdown and uses getRemarkFileDependency when given 1`] = `
"<span
      class=\\"gatsby-resp-image-wrapper\\"
      style=\\"position: relative; display: block; margin-left: auto; margin-right: auto; max-width: 300px; \\"
    >
      <a
    class=\\"gatsby-resp-image-link\\"
    href=\\"not-a-real-dir/not-a-real-dir/images/my-image.jpeg\\"
    style=\\"display: block\\"
    target=\\"_blank\\"
    rel=\\"noopener\\"
  >
    <span
    class=\\"gatsby-resp-image-background-image\\"
    style=\\"padding-bottom: 133.33333333333331%; position: relative; bottom: 0; left: 0; background-image: url('data:image/png;base64,iVBORw'); background-size: cover; display: block;\\"
  ></span>
  <img
        class=\\"gatsby-resp-image-image\\"
        alt=\\"image\\"
        title=\\"\\"
        src=\\"not-a-real-dir/not-a-real-dir/images/my-image.jpeg\\"
        srcset=\\"not-a-real-dir/not-a-real-dir/images/my-image.jpeg, not-a-real-dir/not-a-real-dir/images/my-image.jpeg\\"
        sizes=\\"(max-width: 650px) 100vw, 650px\\"
        style=\\"width:100%;height:100%;margin:0;vertical-align:middle;position:absolute;top:0;left:0;\\"
        loading=\\"lazy\\"
        decoding=\\"async\\"
      />
  </a>
    </span>"
`;
